src/dotnet/UI.Blazor.App/Components/LeftPanel/LeftSearchPanel/RecentlyContactedPeople.razor

```diff
@@ -28,11 +31,24 @@
 
     protected override async Task<IReadOnlyList<Chat>> ComputeState(CancellationToken cancellationToken) {
         var session = Session;
+        var place = Place;
         var recentIds = await ChatUsages.GetRecencyList(session, ChatUsageListKind.PeerChatsWroteTo, cancellationToken).ConfigureAwait(false);
-        var chats = await recentIds
-            .Select(id => Chats.Get(session, id, cancellationToken))
-            .Collect(ApiConstants.Concurrency.High, cancellationToken)
-            .ConfigureAwait(false);
-        return chats.SkipNullItems().ToList();
+        if (ReferenceEquals(place, null)) {
+            var chats = await recentIds
+                .Select(id => Chats.Get(session, id, cancellationToken))
+                .Collect(ApiConstants.Concurrency.High, cancellationToken)
+                .ConfigureAwait(false);
+            return chats.SkipNullItems().ToList();
+        } else {
+            var placeUserIds = await Hub.Places.ListUserIds(session, place.Id, cancellationToken).ConfigureAwait(false);
+            var chats = await recentIds
```

Comment:
I think it's better to filter by place id on backend side. Mark this code with `TODO:`, please